docs/spec/spec.md

```diff
@@ -0,0 +1,152 @@
+5. [UUID Operations](#5-uuid-operations)
+
+## 1. Overview
+This specification elaborates on the functions available in the UUID library.
+
+## 2. Generating UUIDs
+UUID module supports generating 4 versions of UUIDs. They are generated as UUID strings or UUID records. The following is the UUID record definition.
+```
+# Represents a UUID.
+#
+# + timeLow - The low field of the timestamp
+# + timeMid - The middle field of the timestamp
+# + timeHiAndVersion - The high field of the timestamp multiplexed with the version number
+# + clockSeqHiAndReserved - The high field of the clock sequence multiplexed with the variant
+# + clockSeqLo - The low field of the clock sequence
+# + node - The spatially unique node identifier
+public type Uuid readonly & record {
+    ints:Unsigned32 timeLow;
+    ints:Unsigned16 timeMid;
+    ints:Unsigned16 timeHiAndVersion;
+    ints:Unsigned8 clockSeqHiAndReserved;
+    ints:Unsigned8 clockSeqLo;
+    int node;
+};
+```
+
+### 2.1. Version 1
+This is generated using the MAC address of the computer and the time of generation. 
+
+The `uuid:createType1AsString()` function returns a UUID of type 1 as a string.
+```ballerina
+string uuid1 = uuid:createType1AsString();
+```
+
+The `uuid:createType1AsRecord()` function returns a UUID of type 1 as a `Uuid` record.
+```ballerina
+uuid:Uuid uuid1 = check uuid:createType1AsRecord();
+```
+
+### 2.2. Version 3
+This is generated using cryptographic hashing and application-provided text strings. MD5 hashing is used.
+
+UUID library has a set of pre-defined UUIDs strings of well known namespace IDs.
+```
+NAME_SPACE_DNS- Namespace is a fully-qualified domain name
+NAME_SPACE_URL- Namespace is a URL
+NAME_SPACE_OID- Namespace is an ISO OID
+NAME_SPACE_X500- Namespace is an X.500 DN (in DER or a text output format)
+NAME_SPACE_NIL- Empty UUID
+```
+
+The `uuid:createType3AsString()` function returns a UUID of type 3 as a string or else a `uuid:Error`.
+```ballerina
+string uuid3 = check uuid:createType3AsString(uuid:NAME_SPACE_DNS, “ballerina.io”);
+```
+
+The `uuid:createType3AsRecord()` function returns a UUID of type 3 as a `Uuid` record or else a `uuid:Error`.
+```ballerina
+uuid:Uuid uuid3 = check uuid:createType3AsRecord(uuid:NAME_SPACE_DNS, “ballerina.io”);
+```
+
+### 2.3. Version 4
+
+This is generated using a pseudo-random number generator. Every bit of the string is randomly generated.
+
+The `uuid:createType4AsString()` function returns a UUID of type 4 as a string.
+```ballerina
+string uuid4 = uuid:createType4AsString();
+```
+
+The `uuid:createType4AsRecord()` function returns a UUID of type 4 as a `Uuid` record or else a `uuid:Error`.
+```ballerina
+uuid:Uuid uuid4 = check uuid:createType4AsRecord();
+```
+
+### 2.4. Version 5
+This is generated using cryptographic hashing and application-provided text strings. SHA-1 hashing is used.
+
+The `uuid:createType5AsString()` function returns a UUID of type 5 as a string or else a `uuid:Error`.
```

Comment:
Rather than mentioning type5, shall we say
```
The `uuid:createType5AsString()` function returns a UUID of SHA-1 hashing
```
Its much clearer. WDYT?

Reply:
Since there are 2 APIs (`createType5AsString()` and `createType5AsRecord()`) we have mentioned the SHA-1 hashing at the beginning of the section. Checked golang's uuid spec and there in some places the version number is used[[1]](https://pkg.go.dev/github.com/google/uuid#NewUUID) while in other places[[2]](https://pkg.go.dev/github.com/google/uuid#NewMD5) the hashing algorithm is used.

@daneshk thoughts?

[1] https://pkg.go.dev/github.com/google/uuid#NewUUID
[2] https://pkg.go.dev/github.com/google/uuid#NewMD5

Reply:
Shall we just say `The uuid:createType5AsString() function returns a UUID as a string or else a uuid:Error` as we already mentioned it before

Reply:
Fixed in 05798a1